workspaces/vscode/pl-gosu-helper.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# The purpose of this script is to prevent subtle issues that might occur if
+# you are testing in the local PrairieLearn Docker container and it attempts
+# to launch workspaces as root.
+
+# Put a line like "ENV PL_USER rstudio" or "ENV PL_USER coder"
+# (whatever is appropriate) in your Dockerfile. This name should
+# match user 1001.
```

Comment:
I'd like a little more documentation than this, specifically with regards to "whatever is appropriate" - what _is_ appropriate?

Reply:
It's used in `chown -R 1001:1001 "/home/$PL_USER"`, and the intent is that PL_USER should be the name of the expected user 1001 based on however the workspace image is designed.

Reply:
For the workspace images I helped develop, I tried to make sure everything would be 1001:1001 beforehand to prevent slow chowning. But in the case where the local user runs as root, the specific staff user's setup might be creating or mounting stuff into the home tree with different perms, and some things can get messed up if the files in the home tree aren't owned by 1001:1001, so this was a stopgap measure. In particular, I think Dirk ran into this while testing RStudio.

Reply:
...And I'm thinking this may not be necessary with VS Code, since its startup sequence already does the fixuid routine to accomplish something similar. (However, as noted in https://github.com/PrairieLearn/PrairieLearn/issues/3170, relying on fixuid to take care of everything can cause very slow launches, so it's better to make sure as many things as possible are baked into the image as 1001:1001.)